### ISSUE TITLE
Dynamic client

### ISSUE THREAD

**Issue by paralin** [chrisbarrett/kubernetes-el#69]
It's now possible to build a Kubernetes client that learns of all possible resource types from the API, and also is able to access arbitrary resource types by name without a pre-compiled model for the types.

Would it be possible to do the same in kubernetes-el? I'd like to be able to list custom resources.

**Comment by jinnovation** [comment]
+1, the ability to list custom resources would be excellent and expand the use cases for this package significantly.

**Comment by noorul** [comment]
@jinnovation I agree! I thought of using a shared informer to achieve this. It is a matter of free time.

**Comment by jinnovation** [comment]
It's worth noting here that, at least for the problem of a) listing which resource types are available on a given cluster, and b) retrieving resources of a given resource type, a custom Go backend using the shared informer pattern provided by the Kubernetes client package is arguably unnecessary.

For listing resource types, we have:
```bash
kubectl api-resources
```

For fetching resources of arbitrary type `t` (sourced presumably from the former's output), we can simply shell out to the following, like we currently do for the "native" resource types:
```bash
kubectl get t
```

There would naturally be necessary preliminary work to do in the package as a whole to genericize our handling of resources such that hardcodes like [those here](https://github.com/kubernetes-el/kubernetes-el/blob/master/kubernetes-commands.el#L38-L59) are no longer present or necessary. However, in terms of raw primitives necessary to enable the package to know of, retrieve, and display rudimentary information about CRDs, I believe we have the necessary primitives in `kubectl` to do so right now.

**Comment by jinnovation** [comment]
To that effect, I've taken the liberty of renaming this issue to—in my opinion—more accurately reflect the body of work in question. Continuing to call this issue "Dynamic client" would amount to putting the cart before the horse, i.e. describing a solution without first outlining the problem we're trying to solve.

**Comment by noorul** [comment]
@jinnovation Yes, we should make it generic so that we can remove some of the hard codings as you mentioned.

**Comment by jinnovation** [comment]
I've created a [milestone](https://github.com/kubernetes-el/kubernetes-el/milestone/3) to account for all the disparate tasks that need to be taken care of in order to effectively support CRDs.

**Comment by jinnovation** [comment]
We've made some good progress here. See #237, #238, and #239 for details.
